### DISCUSSION TITLE
find_package(imgui)

### DISCUSSION THREAD

**Discussion by grahamreeds** [ocornut/imgui#6486] [New Users Build/Link/Run issues ONLY!]
I am trying to use imgui in a toy ray tracer but the `find_package(imgui REQUIRED)` is failing

> CMake Error at CMakeLists.txt:13 (find_package):
  By not providing "Findimgui.cmake" in CMAKE_MODULE_PATH this project has
  asked CMake to find a package configuration file provided by "imgui", but
  CMake did not find one.

In my Dockerfile I have the following:
```
RUN apt-get update && export DEBIAN_FRONTEND=noninteractive \
    && apt-get -y install --no-install-recommends \
    libglfw3-dev libxrandr-dev libxxf86vm-dev libxcursor-dev libxinerama-dev libxi-dev \
    libglm-dev \
    libimgui-dev \
    libvulkan-dev vulkan-tools vulkan-validationlayers-dev spirv-tools
```

According to #5814 you aren't supposed to do it like this, but I am totally clueless about CMake and can't see how to *simply* consume the imgui library.  My entire CMakeLists.txt file is 23 lines long and 6 lines of that is white space.  Why doesn't it just work like all my other libraries?

**Comment by ocornut** [comment]
> Why doesn't it just work like all my other libraries?

Because it's not like other libraries, we discourage people using precompiled binaries, and for complex reason although it would be trivial to package "imgui", it's not simple at all to package backends in a relevant way. Dear ImGui is not part of any package manager which would be consumed by cmake. 

> Can't see how to simply consume the imgui library.

Compile and link the imgui files the same way you would for your own code.
